### ISSUE TITLE
arrow-return-shorthand rule gives bad suggestion when a function returns an object

### ISSUE THREAD

**Issue by chrismbarr** [palantir/tslint#2134]
### Bug Report

- __TSLint version__: `4.4.1`
- __TypeScript version__: `2.1.5`
- __Running TSLint via__: gulp-tslint 7.0.1

#### TypeScript code being linted

```ts
const getObj = (foo: string) => {
    return {
        "bar": foo,
    };
};
```

I assume it expects something like this, but this **is not valid typescript** 
```ts
const getObj = (foo: string) => {"bar": foo};
```

with `tslint.json` configuration:

```json
"arrow-return-shorthand": true
```

#### Actual behavior

Returns a violation

#### Expected behavior

This should not return a violation since it's valid and there's not another way (that I know of at least) to write this.

**Comment by ajafff** [comment]
It actually expects this
```typescript
const getObj = (foo: string) => ({"bar": foo});
```
which is totally valid typescript and javascript. The automatic fixer also covers this case.

**Comment by chrismbarr** [comment]
Ah ok that makes sense, thank you!

I would have checked the fixer, but VSCode's TSLint no longer automatically works when type checked rules are enabled 👎

**Comment by rokerkony** [comment]
It would still make with some option like in `tslint-eslint-rules`
`"requireReturnForObjectLiteral": true`
